rasa/core/broker.py

```diff
@@ -56,24 +60,109 @@ def publish(self, event: Dict[Text, Any]) -> None:
         raise NotImplementedError("Event broker must implement the `publish` method.")
 
 
+# noinspection PyUnresolvedReferences
+def initialise_pika_connection(
+    host: Text,
+    username: Text,
+    password: Text,
+    connection_attempts: int = 20,
+    retry_delay_in_seconds: Union[int, float] = 5,
+) -> "BlockingConnection":
+    """Create a Pika `BlockingConnection`.
+
+    Args:
+        host: Pika host
+        username: username for authentication with Pika host
+        password: password for authentication with Pika host
+        connection_attempts: number of connection attempts before giving up
+        retry_delay_in_seconds: delay in seconds between connection attempts
+
+    Returns:
+        Pika `BlockingConnection` with provided parameters
+    """
+
+    import pika
+
+    parameters = pika.ConnectionParameters(
+        host,
+        credentials=pika.PlainCredentials(username, password),
+        connection_attempts=connection_attempts,
+        # Wait between retries since
+        # it can take some time until
+        # RabbitMQ comes up.
+        retry_delay=retry_delay_in_seconds,
+    )
+    return pika.BlockingConnection(parameters)
+
+
+# noinspection PyUnresolvedReferences
+def initialise_pika_channel(
+    host: Text, queue: Text, username: Text, password: Text
+) -> "BlockingChannel":
+    """Initialise a Pika channel with a durable queue.
+
+    Args:
+        host: Pika host
+        queue: Pika queue to declare
+        username: username for authentication with Pika host
+        password: password for authentication with Pika host
+
+    Returns:
+        Pika `BlockingChannel` with declared queue
+    """
+
+    connection = initialise_pika_connection(host, username, password)
+
+    return _declare_pika_channel_with_queue(connection, queue)
+
+
+# noinspection PyUnresolvedReferences
+def _declare_pika_channel_with_queue(
+    connection: "BlockingConnection", queue: Text
+) -> "BlockingChannel":
+    """Declare a durable queue on Pika connection."""
+
+    channel = connection.channel()
+    channel.queue_declare(queue, durable=True)
+
+    return channel
+
+
+# noinspection PyUnresolvedReferences
+def close_pika_connection(connection: "BlockingConnection") -> None:
+    """Attempt to close Pika connection."""
+
+    from pika.exceptions import AMQPError
+
+    host = connection.parameters.host
+    try:
+        connection.close()
+        logger.debug("Successfully closed Pika connection with host '{}'.".format(host))
+    except AMQPError:
+        logger.exception("Failed to close Pika connection with host '{}'.".format(host))
```

Comment:
should we include the exception message in the log message?

Reply:
happens automatically with `logger.exception()`

Reply:
Oh cool! Didn't know that 👍